src/test/java/com/ryanharter/auto/value/parcel/AutoValueParcelExtensionTest.java

```diff
@@ -176,43 +256,223 @@
+        "import java.lang.ClassLoader;\n" +
+        "import java.lang.Override;\n" +
+        "import java.lang.String;\n" +
+        "import java.util.List;\n" +
+        "import java.util.Map;\n" +
+        "\n" +
+        "final class AutoValue_Foo extends $AutoValue_Foo {\n" +
+        "  private static final ClassLoader CL = AutoValue_Foo.class.getClassLoader();\n" +
+        "\n" +
+        "  public static final Parcelable.Creator<AutoValue_Foo> CREATOR = new android.os.Parcelable.Creator<AutoValue_Foo>() {\n" +
+        "    @java.lang.Override\n" +
+        "    public AutoValue_Foo createFromParcel(android.os.Parcel in) {\n" +
+        "      return new AutoValue_Foo(in);\n" +
+        "    }\n" +
+        "    @java.lang.Override\n" +
+        "    public AutoValue_Foo[] newArray(int size) {\n" +
+        "      return new AutoValue_Foo[size];\n" +
+        "    }\n" +
+        "  };\n" +
+        "\n" +
+        "  AutoValue_Foo(String a, byte b, int c, short d, long e, float f, double g, boolean h, Parcelable i, CharSequence j, Map<String, String> k, List<String> l, boolean[] m, byte[] n, int[] s, long[] t, Serializable u, SparseArray w, SparseBooleanArray x, Bundle y, PersistableBundle z, Size aa, SizeF ab, Parcelable1 ad, FooBinder ae) {\n" +
+        "    super(a, b, c, d, e, f, g, h, i, j, k, l, m, n, s, t, u, w, x, y, z, aa, ab, ad, ae);\n" +
+        "  }\n" +
+        "\n" +
+        "  private AutoValue_Foo(Parcel in) {\n" +
+        "    super(\n" +
+        "      in.readString(),\n" +
+        "      in.readByte(),\n" +
+        "      in.readInt(),\n" +
+        "      (short) in.readInt(),\n" +
+        "      in.readLong(),\n" +
+        "      in.readFloat(),\n" +
+        "      in.readDouble(),\n" +
+        "      in.readInt() == 1,\n" +
+        "      (Parcelable) in.readParcelable(CL),\n" +
+        "      (CharSequence) in.readCharSequence(),\n" +
+        "      (Map<String, String>) in.readHashMap(CL),\n" +
+        "      (List<String>) in.readArrayList(CL),\n" +
+        "      in.createBooleanArray(),\n" +
+        "      in.createByteArray(),\n" +
+        "      in.createIntArray(),\n" +
+        "      in.createLongArray(),\n" +
+        "      (Serializable) in.readSerializable(CL),\n" +
+        "      in.readSparseArray(CL),\n" +
+        "      in.readSparseBooleanArray(),\n" +
+        "      in.readBundle(CL),\n" +
+        "      in.readPersistableBundle(CL),\n" +
+        "      in.readSize(),\n" +
+        "      in.readSizeF(),\n" +
+        "      (Parcelable1) in.readParcelable(CL),\n" +
+        "      (FooBinder) in.readStrongBinder()\n" +
+        "    );}\n" +
+        "\n" +
+        "  @Override\n" +
+        "  public void writeToParcel(Parcel dest, int flags) {\n" +
+        "    dest.writeString(a());\n" +
+        "    dest.writeInt(b());\n" +
+        "    dest.writeInt(c());\n" +
+        "    dest.writeInt(((Short) d()).intValue());\n" +
```

Comment:
Is the cast and method call actually needed here? The `short` --> `int` implicit conversion should happen